### PR TITLE
Feature/eab 1183

### PR DESCRIPTION
**Pull request by Deepak-Bagrawat**
https://support.cop.homeoffice.gov.uk/browse/EAB-1183

### CODE REVIEW

File: client/src/components/form/DisplayForm.jsx [hunk 1]
```diff
@@ -545,14 +545,15 @@ const DisplayForm = ({
...
-                        <EpmsOfficerEmailsSelect
+                        <EpmsOfficerEmailsAutocomplete
                           id={config.id}
                           fieldId={config.fieldId}
                           required={config.required}
                           onChange={config.onChange}
                           agency={Utils.interpolateString(config.agency, config.formData)}
                           disabled={config.disabled}
-                          value={Utils.interpolateString(config.value, config.formData)}
+                          value={config.formData[config.fieldId]}
```
Comment (LewisClark-HO) [inline client/src/components/form/DisplayForm.jsx:555]: Is there a reason this was changed?

File: client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx [hunk 2]
```diff
@@ -47,29 +57,44 @@ const EpmsOfficerEmailsSelect = ({ id, fieldId, onChange, agency, value, ...attr
...
-    <Select
-      id={id}
-      fieldId={fieldId}
-      options={emails}
-      onChange={onChange}
-      type="select"
-      value={value}
-      {...attrs}
-    />
+    <>
```
Comment (gardnersam) [inline client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx:60]: what is the wrapping fragment for?

File: client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx [hunk 3]
```diff
@@ -10,7 +10,17 @@ const getEmailDomain = (email) => {
...
-const EpmsOfficerEmailsSelect = ({ id, fieldId, onChange, agency, value, ...attrs }) => {
+const EpmsOfficerEmailsAutocomplete = ({
+  id,
+  fieldId,
+  onChange,
+  agency,
+  placeholder,
+  item,
+  value,
+  displayMenu,
```
Comment (gardnersam) [inline client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx:21]: This isn't used either

File: client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx (hunk 2 above)
Comment (Deepak-Bagrawat) [reply client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx:60]: Earlier I was using two component so I added that. I have removed the wrapping fragment.

File: client/src/components/form/DisplayForm.jsx (hunk 1 above)
Comment (Deepak-Bagrawat) [reply client/src/components/form/DisplayForm.jsx:555]: it was not retaining value after selecting from the autocomplete list. now it works as expected

File: client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx [hunk 4]
```diff
@@ -47,29 +55,42 @@ const EpmsOfficerEmailsSelect = ({ id, fieldId, onChange, agency, value, ...attr
...
+EpmsOfficerEmailsAutocomplete.propTypes = {
   id: PropTypes.string.isRequired,
   fieldId: PropTypes.string.isRequired,
-  onChange: PropTypes.func.isRequired,
+  // eslint-disable-next-line react/require-default-props
+  onChange: PropTypes.func,
   agency: PropTypes.string,
   value: PropTypes.string,
+  // eslint-disable-next-line react/require-default-props
+  displayMenu: PropTypes.oneOf(['overlay', 'inline']),
```
Comment (gardnersam) [inline client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx:80]: you've still got things down here that you removed from the props

Comment (Deepak-Bagrawat) [reply client/src/components/Eab/EpmsOfficerEmailsAutocomplete.jsx:80]: yeah, just removed them as well, checking new code

**Review by gardnersam — APPROVED** [review]

**Pull request merged**